sqlx-core/src/postgres/types/ts_vector.rs

```diff
@@ -0,0 +1,50 @@
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::BoxDynError;
+use crate::postgres::{
+    PgArgumentBuffer, PgHasArrayType, PgTypeInfo, PgValueFormat, PgValueRef, Postgres,
+};
+use crate::types::Type;
+
+struct TsVector(String);
+
+impl TsVector {
+    fn from_slice(raw: &[u8]) -> Result<Self, BoxDynError> {
+        let result = std::str::from_utf8(raw)?;
+        Ok(TsVector(result.to_owned()))
+    }
+
+    fn from_str(result: &str) -> Result<Self, BoxDynError> {
+        Ok(TsVector(result.to_owned()))
+    }
+}
+
+impl Type<Postgres> for TsVector {
+    fn type_info() -> PgTypeInfo {
+        PgTypeInfo::TS_VECTOR
+    }
+}
+
+impl PgHasArrayType for TsVector {
+    fn array_type_info() -> PgTypeInfo {
+        PgTypeInfo::TS_VECTOR_ARRAY
+    }
+}
+
+impl Encode<'_, Postgres> for TsVector {
+    fn encode_by_ref(&self, buf: &mut PgArgumentBuffer) -> IsNull {
+        buf.extend_from_slice(self.as_bytes());
+
+        IsNull::No
+    }
+}
+
+impl Decode<'_, Postgres> for TsVector {
+    fn decode(value: PgValueRef<'_>) -> Result<Self, BoxDynError> {
+        match value.format() {
+            PgValueFormat::Binary => TsVector::from_slice(value.as_bytes()?),
```

Comment:
The binary encoding of `tsvector` is unfortunately not this simple: https://github.com/postgres/postgres/blob/8cd61d288adfabe1b7f83359eb0abd27382eec08/src/backend/utils/adt/tsvector.c#L392-L401

And that `WordEntryPos` is the position integer with the weight OR'd into the top two bits: https://github.com/postgres/postgres/blob/a00fd066b1b632e675bae74841a87de1ffc1cd33/src/include/tsearch/ts_type.h#L54-L61

Reply:
Ah... sorry I'll close this and open a feature request I don't think my C is good enough to continue working on this